### PR TITLE
Cleanup little bits of code, consolidate others.

### PR DESCRIPTION
**Pull request by oswynb**
Completed a TODO and made code surrounding it more concise.
Fixed a few minor typos/grammar/layout things.

### CODE REVIEW

File: lib/Network/OAuth2/Server/API.hs [hunk 1]
```diff
@@ -511,39 +503,33 @@ processAuthorizeGet ref user_id permissions response_type client_id' redirect sc
...
-                then return requested_scope
-                else throwError ( Just redirect_uri
-                                , OAuth2Error InvalidScope
-                                              (preview errorDescription "Invalid scope")
-                                              Nothing
-                                )
+        Nothing -> throwInvalidRequest redirect_uri "Scope is missing"
+        Just requested_scope
+            | requested_scope `compatibleScope` permissions -> return requested_scope
+            | otherwise -> throwError
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: Maybe throwInvalidScope as with throwInvalidRequest as used in 507-508 above?

**Comment by thsutton** [comment]
It makes sense to me that we have a throwInvalidScope helper here too. Other than that LGTM, so merge with or without the change.

File: lib/Network/OAuth2/Server/API.hs (hunk 1 above)
Comment (oswynb) [inline lib/Network/OAuth2/Server/API.hs]: Had that originally, but was only used in one place, so I removed it. I'll add it back.

**Comment by christian-marie** [comment]
This is a little ugly now, @thsutton had an idea to split this up a little. I don't see why checkClientAuthCode should return a (return ()). That's pretty bizarre and spaghetti-like in terms of control flow.

**Comment by christian-marie** [comment]
That said, the behaviour is improved and thus merge.

**Pull request merged**